src/RadixDlt.NetworkGateway.Abstractions/Model/PendingTransactionIntentLedgerStatus.cs

```diff
@@ -0,0 +1,136 @@
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+using System;
+
+namespace RadixDlt.NetworkGateway.Abstractions.Model;
+
+/// <summary>
+/// Any additional knowledge we have about the intent.
+/// </summary>
+public enum PendingTransactionIntentLedgerStatus
+{
+    /// <summary>
+    /// We know that the intent has been committed as a success.
+    /// </summary>
+    CommittedSuccess,
+
+    /// <summary>
+    /// We know that the intent has been committed as a failure.
+    /// </summary>
+    CommittedFailure,
+
+    /// <summary>
+    /// We know that the intent has been committed, but don't yet know the outcome until it's synced (Success/Failure).
+    /// </summary>
+    CommitPendingOutcomeUnknown,
+
+    /// <summary>
+    /// We know that the intent is permanently rejected.
+    /// </summary>
+    PermanentRejection,
+
+    /// <summary>
+    /// The intent is known to have been submitted, and at least one payload was not rejected at last submission.
+    /// </summary>
+    PossibleToCommit,
+
+    /// <summary>
+    /// <para>All known payloads containing this intent were rejected at their last submission,
+    /// therefore the intent is likely to end up rejected, but this is not guaranteed.</para>
+    ///
+    /// <para>For example, it could have been rejected because a user didn't have enough XRD in their account
+    /// to lock their fee, and they then top up their account, and the transaction can commit.</para>
+    ///
+    /// <para>Or the payload could have been submitted with the wrong signatures, and the notary corrects
+    /// these and resubmits the intent in a new payload, which can be submitted.</para>
+    /// </summary>
+    LikelyButNotCertainRejection,
+
+    /// <summary>
+    /// We don't know anything about whether the intent could be committed or rejected.
+    /// </summary>
+    Unknown,
+}
+
+public static class PendingTransactionIntentLedgerStatusExtensions
+{
+    /// <summary>
+    /// A new status should overwrite an older status only if it has >= ReplacementPriority.
+    /// </summary>
+    public static int ReplacementPriority(this PendingTransactionIntentLedgerStatus status)
```

Comment:
Holy moly! This function strongly indicates that we must stop trying to override/update any information stored in database and that we need eventsourcing-like database and asychronous state machine for pending TX area. Not that we need to drop RDBMS, just to change the ways of working with this concept.